server/src/main/java/dev/slimevr/tracking/processor/skeleton/LegTweaks.java

```diff
@@ -300,13 +334,20 @@ private boolean preUpdate() {
 			floorLevel = (leftFootPosition.y + rightFootPosition.y) / 2f + FLOOR_CALIBRATION_OFFSET;
 			waistToFloorDist = waistPosition.y - floorLevel;
 
-			// invalidate the buffer since the non initialized output may be
+			// invalidate the buffer since the non-initialized output may be
 			// very wrong
 			bufferInvalid = true;
 			initialized = true;
 		}
 
-		// if not enabled do nothing and return false
+		// update the foot length
+		footLength = skeleton.leftAnkleNode.worldTransform
+			.getTranslation()
+			.distance(
+				skeleton.leftFootNode.worldTransform.getTranslation()
+			);
```

Comment:
skeleton.leftFootNode.localTransform.getTranslation().length should work for that